stacker/actions/diff.py

```diff
@@ -0,0 +1,110 @@
+import logging
+
+from .. import exceptions
+from ..plan import COMPLETE, SKIPPED, Plan
+from . import build
+import difflib
+import json
+import sys
+
+logger = logging.getLogger(__name__)
+
+
+class Action(build.Action):
+    """ Responsible for diff'ing CF stacks in AWS and on disk
+
+    Generates the build plan based on stack dependencies (these dependencies
+    are determined automatically based on references to output values from
+    other stacks).
+
+    The plan is then used to pull the current CloudFormation template from
+    AWS and compare it to the generated templated based on the current
+    config.
+    """
+
+    def _normalize_json(self, json_str, parameters):
```

Comment:
`json_str` should probably template at this point, right?  Also, what was the reasoning behind having the parameters & template normalized into a single string, rather than comparing each separately?  This feels a bit complicated and could be refactored out so that the normalization (and comparison, used in `_diff_stack`) could be easier to read/understand.